manifests/init.pp

```diff
@@ -14,12 +14,18 @@
 # [*node_path*]
 #   Value of the system environment variable (default: "/usr/local/node/node-default/lib/node_modules").
 #
-# [*python_package*]
-#   Python package name, defaults to python
-#
 # [*cpu_cores*]
 #   Number of CPU cores to use for compiling nodejs. Will be used for parallel 'make' jobs.
 #
+# [*contain_ruby*]
+#   Bool flag whether or not to install ruby.
+#
+# [*instances*]
+#   List of instances to install.
```

Comment:
It's basically a hash which contains multiple nodejs versions. In the 1.x versions we had this bloated `nodejs::install` resource. I decided to declare it as private and keep it as implementation detail in order to make changes in the process much easier.
But the feature to install multiple nodejs versions should be kept, so I decided to make it configurable via a hash which means that there's just one entry point right now for the installer.
Furthermore it makes it much easier to setup this module with hiera.